### DISCUSSION TITLE
How to use root model for cookie parameter?

### DISCUSSION THREAD

**Discussion by Danstiv** [fastapi/fastapi#11132] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import Cookie, FastAPI
from pydantic import Field, RootModel

Key = RootModel[Annotated[str, Field(min_length=4)]]


app = FastAPI()

@app.post('/test/')
def test(key: Annotated[Key, Cookie()]):
    pass
```

### Description

I'm trying to use RootModel as a Cookie parameter, but I'm getting the following error:
`AssertionError: Param: key can only be a request body, using Body()`

I'm guessing that it's not that easy to define what exactly a root model is, and it can't be used that way. Therefore, I will describe what I want to achieve.

Our project has a certain str root model with many constraints, a simplified version is used in the code example.

This root model is used as a nested model in several models, and now there is a need to use it in several fastapi endpoints.

So my question is: can I reuse such a root model in both pydantic models and fastapi endpoints?

And if this is not possible, is it possible to implement the same reusable entity without using the RootModel, or is the only solution to duplicate the code?

### Operating System

Linux, Windows

### FastAPI Version

0.109.2

### Pydantic Version

2.5.1

### Python Version

Python 3.12.0

**Comment by jgould22** [comment, marked as answer]
Looks like this is currently an issue with all param types and root model 

https://github.com/tiangolo/fastapi/discussions/11101
